hawkular-inventory-api/src/main/java/org/hawkular/inventory/api/paging/SizeAwarePage.java

```diff
@@ -50,7 +50,7 @@ public SizeAwarePage(Iterator<T> wrapped, PageContext pageContext, HasTotalSize
     @Override
     public long getTotalSize() {
         if (totalSize == HasTotalSize.NOT_DEPLETED && !hasNext()) {
-            totalSize = hasTotalSize.getTotalSize();
+            totalSize = hasTotalSize == null ? HasTotalSize.NOT_DEPLETED : hasTotalSize.getTotalSize();
```

Comment:
> it's just that `hasTotalSize` has already been nulled out (which I guess can only happen when you somehow call `close()` and `hasNext()` still returns true).

Yes, that was the case, the close() was called prematurely. I fixed it so it should work also w/o this check, I just wanted to be defensive here not allowing to throw the NPE. Perhaps checking the hasTotalSize for null and throwing IllegalStateExc. is better.

Now, I see the commit message is wrong, it should say: "issue resulting in calling Page.getTotalSize() on non-depleted but closed page -> NPE"